mapswipe_workers/mapswipe_workers/utils/tile_grouping_functions.py

```diff
@@ -291,6 +299,53 @@ def groups_intersect():
             and (int(y_minB) <= int(y_max))
         )
 
+    def merge_groups():
```

Comment:
I would suggest making this a pure function, that takes 2 groups (or whatever you need) as arguments and returns a new group, and does not use any external variables. Right now, you're relying on context to give you the input data, and this tends to be very fragile, and therefore bug prone. If you move this function elsewhere in your code, or you change your code before or after this, you might change the meaning of a variable, and have to update the body of this function. You're breaking the "encapsulation" of this function in doing so, and it often leads to sad developers :)